File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/domain/repository/PublishingHouseRepository.java
package ru.elenacherev.librarymanager.domain.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import ru.elenacherev.librarymanager.domain.entity.PublishingHouseEntity;

@Repository
@Transactional(propagation = Propagation.MANDATORY)
public interface PublishingHouseRepository extends JpaRepository<PublishingHouseEntity, Long> {
}

File: librarymanager-be/src/main/kotlin/ru/elenacherev/librarymanager/domain/repository/PublishingHouseRepository.kt
package ru.elenacherev.librarymanager.domain.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import ru.elenacherev.librarymanager.domain.entity.PublishingHouseEntity

@Repository
@Transactional(propagation = Propagation.MANDATORY)
interface PublishingHouseRepository : JpaRepository<PublishingHouseEntity, Long> 